libnmstate/state.py

```diff
@@ -238,6 +238,17 @@ def verify_routes(self, other_state):
                             Route.KEY: other_routes
                         }))
 
+    def verify_dns(self, other_state):
+        if self.config_dns != other_state.config_dns:
+            raise NmstateVerificationError(
```

Comment:
It will be nice to have an inherited error that mentions this is a dns failure. The you can use it in the tests nicely, also at the integration level.
(not a must)

Reply:
There is no precedent for verification error yet.
We can discuss that in another PR.